### PR TITLE
feat: projects

### PR DESCRIPTION
**Pull request by Onitsiky**

### CODE REVIEW

File: doc/api.yml [hunk 1]
```diff
@@ -125,6 +125,90 @@ paths:
...
+      operationId: getAllProjects
+      parameters:
+        - in: query
+          name: page
+          schema:
+            $ref: '#/components/schemas/Page'
+        - in: query
+          name: pageSize
+          schema:
+            $ref: '#/components/schemas/PageSize'
```
Comment (Mahefaa) [inline doc/api.yml:148]: add filter by userId

File: doc/api.yml [hunk 2]
```diff
@@ -125,6 +125,90 @@ paths:
...
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Project'
+        400:
+          $ref: "#/components/responses/400"
+        500:
+          $ref: "#/components/responses/500"
+        404:
+          $ref: "#/components/responses/404"
```
Comment (Mahefaa) [inline doc/api.yml]: no possible 404 for a getAll

File: doc/api.yml [hunk 3]
```diff
@@ -125,6 +125,90 @@ paths:
...
+          required: true
+      responses:
+        200:
+          description: The Project
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Project'
+        400:
+          $ref: "#/components/responses/400"
```
Comment (Mahefaa) [inline doc/api.yml]: no possible 400 for this

Comment (Mahefaa) [reply doc/api.yml]: however 403 is possible

File: doc/api.yml [hunk 4]
```diff
@@ -125,6 +125,90 @@ paths:
...
+      summary: Create or update projects
+      operationId: crupdateProjects
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: array
+              items:
+                $ref: '#/components/schemas/CreateProject'
+      responses:
```
Comment (Mahefaa) [inline doc/api.yml:174]: add 403

File: doc/api.yml [hunk 5]
```diff
@@ -229,4 +319,41 @@ components:
...
+        health:
+          $ref: '#/components/schemas/ProjectHealth'
+    Project:
+      type: object
+      allOf:
+        - $ref: '#/components/schemas/CreateProject'
+      properties:
+        id:
+          type: string
+
```
Comment (Mahefaa) [inline doc/api.yml]: put to CreateProject

File: doc/api.yml [hunk 6]
```diff
@@ -125,6 +125,90 @@ paths:
           $ref: "#/components/responses/500"
         404:
           $ref: "#/components/responses/404"
+  /projects:
+    get:
+      tags:
+        - Projects
+      security:
+        - [ ]
```
Comment (Mahefaa) [inline doc/api.yml]: no security is put like this : 
``
      security: [ ]
``

**Review by Mahefaa — CHANGES_REQUESTED** [review]

File: doc/api.yml (hunk 1 above)
Comment (Onitsiky) [reply doc/api.yml:148]: Filtered by ownerId

**Pull request merged**